master/internal/api/apiutils/utils.go

```diff
@@ -47,6 +47,8 @@ func MapAndFilterErrors(err error) error {
 		return status.Error(codes.NotFound, err.Error())
 	case errors.Is(err, db.ErrDuplicateRecord):
 		return status.Error(codes.AlreadyExists, err.Error())
+	case errors.Is(err, db.ErrNotEnoughPermissions):
+		return err
```

Comment:
I think this should be ```status.Error(codes.PermissionDenied, err.Error()``` right?

Reply:
The db error has been changed to have the status code in it, so the error should be that already.